nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -320,8 +320,17 @@ bool RegulatedPurePursuitController::shouldRotateToGoalHeading(
   const geometry_msgs::msg::PoseStamped & carrot_pose)
 {
   // Whether we should rotate robot to goal heading
-  double dist_to_goal = std::hypot(carrot_pose.pose.position.x, carrot_pose.pose.position.y);
-  return params_->use_rotate_to_heading && dist_to_goal < goal_dist_tol_;
+  double dist_to_goal =
+    std::hypot(carrot_pose.pose.position.x, carrot_pose.pose.position.y);
+
+  if (params_->stateful) {
```

Comment:
I tihnk this can all be simplified a bit.

At the start of the function 

```
if (!params_->use_rotate_to_heading) {
  return false;
}
```

That removes the need for that in any of the other areas below. We can then simply the rest I think more easily

Reply:
I add your suggestion